go/appencryption/README.md

```diff
@@ -0,0 +1,257 @@
+if err != nil {
+	panic(err)
+}
+```
+
+A more extensive example is the [Reference Application](cmd/example/README.md), which will evolve along 
+with the SDK.
+
+## How to Use Asherah
+
+Before you can start encrypting data, you need to define Asherah's required pluggable components. Below we show how to
+build the various options for each component.
+
+### Define the Metastore
+
+Detailed information about the Metastore, including any provisioning steps, can be found [here](../../docs/Metastore.md).
+
+#### RDBMS Metastore
+
+Asherah can connect to a relational database by accepting a connection string and opening a connection to the database
+using a database driver name. See https://golang.org/s/sqldrivers for a list of third-party drivers.
+
+```go
+// Open a DB connection using a database driver name and connection string
+connectionString := ...;
+
+// Parse the DSN string to a Config
+dsn, err := mysql.ParseDSN(connectionString)
+if err != nil {
+    return err
+}
+
+// Open a connection to the database using the driver name and the connection string
+db, err := sql.Open("driver-name", dsn.FormatDSN())
+if err != nil {
+    return err
+}
+
+// Build the Metastore
+metastore := persistence.NewSQLMetastore(db)
+```
+
+#### DynamoDB Metastore
+
+```go
+sess, err = session.NewSession(&aws.Config{
+    Region: aws.String("us-west-2"), // specify preferred region here
+})
+if err != nil {
+    panic(err)
+}
+// Build the Metastore
+metastore := persistence.NewDynamoDBMetastore(sess)
+```
+
+#### In-memory Metastore (FOR TESTING ONLY)
+
+```go
+metastore := persistence.NewMemoryMetastore()
+```
+
+### Define the Key Management Service
+Detailed information about the Key Management Service can be found [here](../../docs/KeyManagementService.md).
+
+#### AWS KMS
+
+```go
+// Create a map of region and ARN pairs that will all be used when creating a System Key
+regionArnMap := map[string]string
+{
+    "us-west-2": "ARN FOR US-WEST-2",
+    "us-east-2": "ARN FOR US-EAST-2",
+    "eu-west-2": "ARN FOR EU-WEST-2",
+    ...,
+}
+crypto := aead.NewAES256GCM()
+
+// Build the Key Management Service using the region dictionary and your preferred (usually current) region
+keyManagementService :=  kms.NewAWS(crypto, "us-west-2", regionArnMap)
+```
+
+#### Static KMS (FOR TESTING ONLY)
+
+```go
+crypto := aead.NewAES256GCM()
+keyManagementService := kms.NewStatic("mysupersecretstaticmasterkey!!!!", crypto)
+```
+
+### Define the Crypto Policy
+Detailed information on Crypto Policy can be found [here](../../docs/CryptoPolicy.md). The Crypto Policy's effect 
+on key caching is explained [here](../../docs/KeyCaching.md).
+
+#### Basic Expiring Crypto Policy
+
+```go
+cryptoPolicy := appencryption.NewCryptoPolicy()
+```
+
+The default key expiration limit is 90 days and revoke check interval is 60 minutes. These can be changed using 
+functional options. The unit of `key expiration limit` is `days` while the unit of `revoke check interval` is `minutes`
```

Comment:
go implementation doesn't have default units... well I suppose it does but it would be `nanoseconds` and not `days` and `minutes` noted here.

Reply:
Decided to remove the `unit` line altogether. Updated `policy.go` to use 90 days and 60 minutes as per the docs.